### ISSUE TITLE
Documentation on building and usage for appropriate architecture

### ISSUE THREAD

**Issue by alloydwhitlock** [firehydrant/terraform-provider-firehydrant#2]
**Problem**
The provider is built for macOS, which will cause issues if using `linux_amd64`. You will receive an error stating `Error: Incompatible provider version` when attempting to use it in Linux following the basic "Use Provider" instructions. 

**Solution**
- Provide directions on how to build architecture-dependent version for Linux, Windows, or macOS. 
- List provided TF provider binary is macOS (`Mach-O 64-bit`)

**Comment by dcondomitti** [comment]
I can't replicate this after copying the example provider block from the documentation site. I'm not super familiar with how Hashicorp stores the provider binaries but I see one for linux_amd64. Is there a chance your `.terraform` directory was originally populated on a Mac? 

This was the binary fetched when I ran `terraform init`:

```
/.terraform/providers/registry.terraform.io/firehydrant/firehydrant/0.1.2/linux_amd64 # file terraform-provider-firehydrant_v0.1.2
terraform-provider-firehydrant_v0.1.2: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, Go BuildID=kiAiGbFYQOxPcrp6pMhq/R6ORj9x7ha2Qt2XmEyV2/zRhO_jHMxq6RHE6NEq-6/T16BT2kRP_FP8qmSwsg_, stripped

**Comment by alloydwhitlock** [comment]
I used `TF_LOG=DEBUG` to determine that it was sourcing a cached version from my `~/.terraform.d` directory for the provider. I removed the directory and ran `terraform init`, which worked. 

Digging into what caused my initial environment issue a bit further and found it started with running the `quick-redo.sh` script in the examples directory. Why did I run the `quick-redo.sh` script? It looked innocuous enough when looking at it, which was why I did it.  `quick-redo.sh` has a `make install` step, which has a variable set for `OS_ARCH=darwin_amd64`. 

Mystery solved.
